ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -92,9 +92,12 @@ function corsMiddleware(options, next) {
 function apiPathModifierMiddleware(options, next) {
 	const { siteApiNamespace } = getGBKit();
 
-	if (siteApiNamespace) {
-		// Insert the API namespace after the first two path segments. 
-		options.path = options.path.replace(/^((?:\/[\w.-]+){2})/, `$1/${siteApiNamespace}`);
+	if (siteApiNamespace && !options.path.includes(siteApiNamespace)) {
```

Comment:
In what context is the namespace already present?

Reply:
I'll test it again and report back, but I remember I started to have issues once we merged this code and I was playing around with some endpoints, where it was appending the namespace at least twice 😅

Reply:
By just having the API credentials set, as soon as the editor loads with these changes I'm seeing failures to:

`https://public-api.wordpress.com/wp/v2/sites/SITE_ID/sites/SITE_ID/blocks?per_page=100&_locale=user`

It might be an issue related to how the reusable blocks are fetched, but I'm not sure if it would happen with other endpoints. I'd say to include this check. What do you think?

Reply:
Interesting. I'm curious as to why, but I can research further later. Yes, I agree including the check makes sense.